### ISSUE TITLE
TS & next-routes doesn't play well together.

### ISSUE THREAD

**Issue by MathiasKandelborg** [zeit/next.js#5516]
# Bug report

I'm having a lot of trouble getting TS and next-routes working together. The last time I had a successful production build, it was on version 6. [I created a thread on spectrum](https://spectrum.chat/?t=80b77c79-752a-4d72-8e67-50ff5c35af7c) asking for help, then realized a reproducible example would be appropriate. So, I took the custom server TS example which I've based my current project on and added next-routes, which resulted in an actual error in dev mode! 

## Describe the bug

I'm not sure what's wrong. It's probably something with babel, I'll attach the full error here.

The actual thing I was trying to get help about does not yield any errors or output whatsoever. Building the app is no problem. Running the build without production mode yields a `404` on _all_ pages. 

```bash
ERROR  Failed to compile with 1 errors                                                                                                                                                        17:46:19

 error  in ./pages/index.tsx

Module parse failed: Unexpected token (5:2)
You may need an appropriate loader to handle this file type.
|
| export default () => (
>   <ul>
|     <li><Link href='/a' as='/a'><a>a</a></Link></li>
|     <li><Link href='/b' as='/b'><a>b</a></Link></li>

 @ multi ./pages/index.tsx
```

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:
1. Go to https://github.com/MathiasKandelborg/custom-server-typescript-app
1. Clone the repo
1. Install with yarn/npm
1. Run the app in dev mode
1. See error

## Expected behavior

The routing specified in routes.js works. Furthermore that no babel errors show up. 

## System information

 - OS: Ubuntu 18.04 - all updated
 - Browser latest Google Chrome & FF (no differences)
 - Version of Next.js: 7.0.2

**Comment by echoulen** [comment]
missing [.babelrc](https://github.com/zeit/next.js/blob/canary/examples/custom-server-typescript/.babelrc)

**Comment by MathiasKandelborg** [comment]
The babelrc doesn't make a difference in this case. I've just forgotten to re-add it, before commiting.

Also, I've got everything working with just next routing. Turns out that next-routes aren't needed at all for most cases. Next is really fantastic.

So the issue still persists and I have not gotten further in troubleshooting. I have simply discarded next-routes for now.

I will take a stab at this again once I've caught up with my (now working!) initial deployment and the first ŕound of bug fixing that goes with that.

**Comment by echoulen** [comment]
After modifying your case, it should be work

server.ts
```
import routes from "./routes";
...
```

routes.ts

```
const nextRoutes = require('next-routes');
const routes = nextRoutes()
    .add('index', '/')
    .add('a', '/a', 'a')
    .add('b', '/b', 'a');

export const Link = routes.Link;
export default routes;
```

index.tsx
```
import React from 'react'
import {Link} from '../server/routes'

export default () => (
  <ul>
    <li><Link href='/a' as='/a'><a>a</a></Link></li>
    <li><Link href='/b' as='/b'><a>b</a></Link></li>
  </ul>
)
```

**Comment by MathiasKandelborg** [comment]
I'm sorry for the wait!

I've updated the test repo with two commits.
- The first commit implements the suggestion from @echoulen. The app still wouldn't work, with the exact same behavior.
- The second commit contains the current fix I've found, commenting out the custom config in the next function in `/server/server.ts`.

I'm not sure if this is the expected behavior?
I'd like to think that all that's needed are some lines of code in the custom config.
